webportal/src/app/job/job-view/job-view.component.js

```diff
@@ -147,20 +147,45 @@ const loadJobs = () => {
       if (data.error) {
         alert(data.message);
       } else {
-        $('#view-table').html(jobTableComponent({
-          jobs: data.slice(0, 300),
-          getDurationInSeconds,
-          convertTime,
-          convertState,
-        }));
+        let displayDataSet = [];
+        let rowCount = limit ? limit : 1000;
+        for (let i = 0; i < data.slice(0, rowCount).length; i++) {
```

Comment:
use `rowCount` instead of `data.slice(0, rowCount).length`, `rowCount` should be `Math.min(data.length, limit)`